### DISCUSSION TITLE
TSC compiler errors with slot(s)

### DISCUSSION THREAD

**Discussion by jzs11** [vuejs/language-tools#551] [Q&A] [answered]
Hi dude,

SOS.

I have upgrade everything to latest

vue-tsc: "^3.0.0"
vite: "^2.5.0"

When I do a publish it gives me few errors around the slot/template area
![image](https://user-images.githubusercontent.com/2770401/135767806-b13295c7-5a1d-4060-979c-da657b02e8c5.png)

![image](https://user-images.githubusercontent.com/2770401/135767817-c6d8fdd7-8154-4bb2-adf7-448a271b7475.png)

Here is the d.ts file for select button
![image](https://user-images.githubusercontent.com/2770401/135767830-5241ff6f-f1d9-4de6-aa67-9125c8e9f932.png)

Those errors are not showing before, possible give me an idea or direction where I should ahead?

Much appreciated!
Jing

**Comment by jzs11** [comment, marked as answer]
hmm I spent the whole morning to fix this thing. It turns out if I remove the lock.json file then do a npm i, the problem will come to bite me.

But If I use:
- npm uninstall vue-tsc
- npm clear cache --force
- npm install vue-tsc

All those errors are gone. 

However, one thing I noticed those local import components will show type any instead of proper type.

![image](https://user-images.githubusercontent.com/2770401/135773890-13786814-1037-40b7-bc65-b122d10f8140.png)

**Reply by jzs11** [reply]
@johnsoncodehk I think this problem could be due to some vue components do not need to have script tag. Such as..

![image](https://user-images.githubusercontent.com/2770401/136719228-8afcba2e-3b33-4e1c-89cf-dd80b7720cf2.png)

Then it will lead to Any type

F.y.i

**Reply by johnsoncodehk** [reply]
This is designed behavior to fix #412, because vue 3 `DefineComponent` type is not compatible to `@vue/composition-api` and `vue-class-component`, we can't hypothesis component is `DefineComponent` for no script block SFC.

But I think we have some way to keep slots type, you can open a issue to track.

**Reply by jzs11** [reply]
I think you are too busy.. it does not impact me a lot. just for your info. Thanks for the effort to maintain this great plugin.
